### ISSUE TITLE
The parser should support function names such as new, list, etc

### ISSUE THREAD

**Issue by bbtgp** [glayzzle/php-parser#91]
Currently this throws an exception SyntaxError: Parse Error : syntax error, unexpected 'new' (T_NEW), expecting T_STRING on line 1

Changing this line in read_function_declaration/function.js:79 fixes the problem for me
```javascript
if (this.token === this.tok.T_NEW || this.token === this.tok.T_LIST || this.expect(this.tok.T_STRING)) {

```
I can make a pull request unless you want something cleaner?
Cheers!

**Comment by ichiriac** [comment]
Hi  @bbtgp,

Thanks for the report, I knowed this bug a while but thought it will be introduced in PHP 7.2. Here the full specification regarding your request : https://wiki.php.net/rfc/context_sensitive_lexer

Here the implementation of the RFC : https://github.com/php/php-src/blob/master/Zend/zend_language_parser.y#L284

The idea is to define here a new list `IDENTIFIER` of possible tokens : https://github.com/glayzzle/php-parser/blob/master/src/parser.js#L36

Next, you need to replace `this.token === this.tok.T_STRING` with `this.is('IDENTIFIER')` at every place where PHP uses them by searching the identifier keyword in their parser (like here https://github.com/php/php-src/blob/master/Zend/zend_language_parser.y#L725)

In order to keep the coverage level clean, on each change you need to introduce a new test.

As this is part of PHP 7.0 and `php-parser` should cover it fully, it's actually a critical bug. 

Please tell me if you think you can find time to implement it, or I'll try to pass this myself a bit later.

**Comment by bbtgp** [comment]
#92

**Comment by ichiriac** [comment]
Hi @bbtgp, it's now released under 2.1.0. 

Many thanks for your contribution !
